bindings/python/iota_sdk/types/payload.py

```diff
@@ -33,12 +38,94 @@ class TaggedDataPayload:
         tag: The tag part of the tagged data payload.
         data: The data part of the tagged data payload.
     """
-    tag: HexStr
-    data: HexStr
     type: int = field(
         default_factory=lambda: int(
             PayloadType.TaggedData),
         init=False)
+    tag: HexStr
+    data: HexStr
+
+
+def deserialize_payload(d: Dict[str, Any]) -> Payload:
+    """Takes a dictionary as input and returns an instance of a specific class based on the value of the 'type' key in the dictionary.
+
+    Arguments:
+    * `d`: A dictionary that is expected to have a key called 'type' which specifies the type of the returned value.
+    """
+    payload_type = d['type']
+    if payload_type == PayloadType.TaggedData:
+        return TaggedDataPayload.from_dict(d)
+    if payload_type == PayloadType.SignedTransaction:
+        return SignedTransactionPayload.from_dict(d)
+    if payload_type == PayloadType.CandidacyAnnouncement:
+        return CandidacyAnnouncementPayload.from_dict(d)
+    raise Exception(f'invalid payload type: {payload_type}')
+
+
+def deserialize_payloads(
+        dicts: List[Dict[str, Any]]) -> List[Payload]:
+    """Takes a list of dictionaries as input and returns a list with specific instances of classes based on the value of the 'type' key in the dictionary.
+
+    Arguments:
+    * `dicts`: A list of dictionaries that are expected to have a key called 'type' which specifies the type of the returned value.
+    """
+    return list(map(deserialize_payload, dicts))
+
+
+def deserialize_tagged_data_payload(d: Dict[str, Any]) -> Payload:
+    """Takes a dictionary as input and returns an instance of a specific class based on the value of the 'type' key in the dictionary.
+
+    Arguments:
+    * `d`: A dictionary that is expected to have a key called 'type' which specifies the type of the returned value.
+    """
+    payload_type = d['type']
+    if payload_type == PayloadType.TaggedData:
+        return TaggedDataPayload.from_dict(d)
+    raise Exception(f'invalid payload type: {payload_type}')
+
+
+@json
+@dataclass
+class Transaction:
```

Comment:
Why didn't you leave this in transaction.py?

Reply:
Because the cyclic import fix didn't work anymore with the changes. It might have changed now with the latest changes. Want me to try to bring it back?

Reply:
Yes please

Reply:
Had a quick look myself and it's not just about the typings which could be solved with `TYPE_CHECKING`, maybe there is no proper way then, will approve as it is